ddl/backfilling.go

```diff
@@ -232,35 +288,64 @@ func mergeBackfillCtxToResult(taskCtx *backfillTaskContext, result *backfillResu
 }
 
 type backfillWorker struct {
-	id        int
-	reorgInfo *reorgInfo
-	batchCnt  int
-	sessCtx   sessionctx.Context
-	taskCh    chan *reorgBackfillTask
-	resultCh  chan *backfillResult
-	table     table.Table
-	priority  int
-	tp        backfillerType
-	ctx       context.Context
-	cancel    func()
+	id int
+	backfiller
+	taskCh   chan *reorgBackfillTask
+	resultCh chan *backfillResult
+	ctx      context.Context
+	cancel   func()
 }
 
-func newBackfillWorker(ctx context.Context, sessCtx sessionctx.Context, id int, t table.PhysicalTable,
-	reorgInfo *reorgInfo, tp backfillerType) *backfillWorker {
+func newBackfillWorker(ctx context.Context, id int, bf backfiller) *backfillWorker {
 	bfCtx, cancel := context.WithCancel(ctx)
 	return &backfillWorker{
-		id:        id,
-		table:     t,
-		reorgInfo: reorgInfo,
-		batchCnt:  int(variable.GetDDLReorgBatchSize()),
-		sessCtx:   sessCtx,
-		priority:  reorgInfo.Job.Priority,
-		tp:        tp,
-		ctx:       bfCtx,
-		cancel:    cancel,
+		backfiller: bf,
+		id:         id,
+		taskCh:     make(chan *reorgBackfillTask, 1),
+		resultCh:   make(chan *backfillResult, 1),
+		ctx:        bfCtx,
+		cancel:     cancel,
 	}
 }
 
+func (w *backfillWorker) updateLease(execID string, bJob *BackfillJob) error {
+	isDistReorg, err := w.GetCtx().sessCtx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(variable.TiDBDDLEnableDistributeReorg)
```

Comment:
To prevent the concurrent updates on this variable, maybe we can persist it to `model.Job.ReorgMeta.ReorgType` and keep it unchanged once it is determined.